features/networking/sriov.feature

```diff
@@ -66,6 +69,806 @@ Feature: Sriov related scenarios
+    Then the step should succeed
+    And the output should contain:
+      | <cardname>-netdevice  |
+      | syncStatus: Succeeded |
+    """
+    Given I switch to the first user
+    And I have a project
+    And evaluation of `project.name` is stored in the :usr_project clipboard
+    Given I obtain test data file "networking/sriov/sriovnetwork/<cardname>netdevice.yaml"
+    Given I create sriov resource with following:
+      | cr_yaml       | <cardname>netdevice.yaml |
+      | cr_name       | <cardname>-netdevice     |
+      | resource_type | sriovnetwork             |
+      | project       | <%= cb.usr_project%>     |
+    Then the step should succeed
+
+    And I wait up to 30 seconds for the steps to pass:
+    """
+    When I run the :get admin command with:
+      | resource  | net-attach-def        |
+      | namespace | <%= cb.usr_project%>  |
+    Then the step should succeed
+    And the output should contain "<cardname>-netdevice"
+    """
+    And I use the "<%= cb.usr_project%>" project
+    Given I obtain test data file "networking/sriov/pod/sriov-macvlan.yaml"
+    When I run oc create over "sriov-macvlan.yaml" replacing paths:
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"] | <cardname>-netdevice |
+    Then the step should succeed
+    And a pod becomes ready with labels:
+      | name=sriov-macvlan |
+    When I execute on the pod:
+      | /usr/sbin/ip | a |
+    Then the output should contain:
+      | 10.56.217 |
+
+    Examples:
+      | cardname | 
+      | mlx277   | # @case_id OCP-24774
+      | mlx278   | # @case_id OCP-24775
+
+  # @author zzhao@redhat.com
+  # @case_id OCP-24780
+  @destructive
+  @admin  
+  Scenario: NAD will be deleted too when sriovnetwork is deleted
+    Given the sriov operator is running well
+    Given I switch to the first user
+    And I have a project
+    And evaluation of `project.name` is stored in the :usr_project1 clipboard
+    Given I obtain test data file "networking/sriov/sriovnetwork/intelnetdevice.yaml"
+    Given I create sriov resource with following:
+      | cr_yaml       | intelnetdevice.yaml   |
+      | cr_name       | intel-netdevice-rhcos |
+      | resource_type | sriovnetwork          |
+      | project       | <%= cb.usr_project1%> |
+    Then the step should succeed
+
+    And I wait up to 30 seconds for the steps to pass:
+    """
+    When I run the :get admin command with:
+      | resource  | net-attach-def        |
+      | namespace | <%= cb.usr_project1%> |
+    Then the step should succeed
+    And the output should contain "intel-netdevice-rhcos"
+    """
+    Given I delete the "intel-netdevice-rhcos" sriovnetwork
+    And I wait up to 30 seconds for the steps to pass:
+    """
+    When I run the :get admin command with:
+      | resource  | net-attach-def        |
+      | namespace | <%= cb.usr_project1%> |
+    Then the step should succeed
+    And the output should not contain "intel-netdevice-rhcos"
```

Comment:
We can use step `Given /^the( admin)? (\w+) named #{QUOTED} does not exist(?: in the#{OPT_QUOTED} project)?$/ do |who, resource_type, resource_name, project_name|`

Reply:
updated